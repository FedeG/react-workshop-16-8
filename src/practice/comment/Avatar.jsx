import React from 'react';
import PropTypes from 'prop-types';

export default class Avatar extends React.PureComponent {
  static propTypes = {
    author: PropTypes.shape({
      avatarUrl: PropTypes.string,
      name: PropTypes.string,
    }).isRequired,
    onClickHandler: PropTypes.func.isRequired,
  };

  handleOnClick = (event) => {
    event.preventDefault();
    event.stopPropagation();
    const { onClickHandler } = this.props;
    onClickHandler();
  };

  render() {
    const { author: { avatarUrl, name } } = this.props;
    return (
      <div
        role="button"
        tabIndex={0}
        onKeyPress={this.handleOnClick}
        onClick={this.handleOnClick}
      >
        <img className="Avatar" src={avatarUrl} alt={name} />
      </div>
    );
  }
}
