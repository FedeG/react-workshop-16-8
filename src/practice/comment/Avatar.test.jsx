import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './Avatar';

describe('Avatar', () => {
  let component;
  let onClickHandler;

  const author = {
    name: 'Test',
    avatarUrl: 'avatarTestUrl',
  };

  beforeEach(() => {
    onClickHandler = jest.fn();
    component = renderer.create(
      <Avatar
        author={author}
        onClickHandler={onClickHandler}
      />,
    );
  });

  test('initial state', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('onClickHandler should be called when user click on avatar', () => {
    const instance = component.getInstance();
    const event = {
      preventDefault: jest.fn(),
      stopPropagation: jest.fn(),
    };
    instance.handleOnClick(event);

    expect(event.preventDefault).toBeCalled();
    expect(event.stopPropagation).toBeCalled();
    expect(onClickHandler).toBeCalled();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
