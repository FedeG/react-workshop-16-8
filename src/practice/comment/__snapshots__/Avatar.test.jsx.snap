// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar initial state 1`] = `
<div
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <img
    alt="Test"
    className="Avatar"
    src="avatarTestUrl"
  />
</div>
`;

exports[`Avatar onClickHandler should be called when user click on avatar 1`] = `
<div
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <img
    alt="Test"
    className="Avatar"
    src="avatarTestUrl"
  />
</div>
`;
